### DISCUSSION TITLE
Teleport support ?

### DISCUSSION THREAD

**Discussion by mgnrfk** [nuxt/nuxt#18040] [Questions] [answered]
How can we use teleport without mounting or hydration issues?

It would be great to have a simple way to use vue 3 teleport feature.
Maybe allow adding targets out of the layout scope, or a nuxt-portal module to have an out of the box solution?

**Comment by lucpotage** [comment, marked as answer]
There is at least the new `<NuxtLayout>` component that allows you to define named layout slots. Would it be sufficient in your case? https://v3.nuxtjs.org/docs/directory-structure/layouts#example-using-with-slots

**Reply by mgnrfk** [reply]
Actualy even the default layout method works for my use case, I just didn't thought of using it for my tests.. Shame on me!

**Reply by wobsoriano** [reply]
But yeah, it would be nice to have teleport support without mounting/hydration issues!

**Comment by elHornair** [comment]
Depending on what exactly you're teleporting, you can also wrap the `<teleport>` in a `<ClientOnly>` component (https://v3.nuxtjs.org/docs/directory-structure/components#clientonly-component).
In my case (rendering a Modal), this solved the issue just fine, because I only want the Modal to teleport after a user action anyway (so only on the client side).
